baseapp/abci.go

```diff
@@ -550,12 +576,108 @@ func (app *BaseApp) ProcessProposal(_ context.Context, req *abci.RequestProcessP
 
 	resp, err = app.processProposal(app.processProposalState.ctx, req)
 	if err != nil {
-		return nil, err
+		app.logger.Error("failed to process proposal", "height", req.Height, "error", err)
+		return &abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}, nil
 	}
 
 	return resp, nil
 }
 
+// ExtendVote implements the ExtendVote ABCI method and returns a ResponseExtendVote.
+// It calls the application's ExtendVote handler which is responsible for performing
+// application-specific business logic when sending a pre-commit for the NEXT
+// block height. The extensions response may be non-deterministic but must always
+// be returned, even if empty.
+//
+// Agreed upon vote extensions are made available to the proposer of the next
+// height and are committed in the subsequent height, i.e. H+2. An error is
+// returned if vote extensions are not enabled or if extendVote fails or panics.
+func (app *BaseApp) ExtendVote(_ context.Context, req *abci.RequestExtendVote) (resp *abci.ResponseExtendVote, err error) {
+	// Always reset state given that ExtendVote and VerifyVoteExtension can timeout
+	// and be called again in a subsequent round.
+	emptyHeader := cmtproto.Header{ChainID: app.chainID, Height: req.Height}
+	app.setState(execModeVoteExtension, emptyHeader)
+
+	// If vote extensions are not enabled, as a safety precaution, we return an
+	// error.
+	cp := app.GetConsensusParams(app.voteExtensionState.ctx)
+	if cp.Abci != nil && cp.Abci.VoteExtensionsEnableHeight <= 0 {
```

Comment:
Do we need a consensus module migration here? Or can we capture this somehow so we don't forget

Reply:
An app would have to set the `VoteExtensionsEnableHeight` in it's upgrade handler, yes. I dont think this is something we can do for them. That being said, you mean should we set default values in `x/consensus`?